device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -267,64 +291,91 @@ public void onConnectionEstablished(String connectionId)
     {
         if (connectionId.equals(this.iotHubTransportConnection.getConnectionId()))
         {
-            log.info("The connection to the IoT Hub has been established");
+            log.debug("The connection to the IoT Hub has been established");
 
-            //Codes_SRS_IOTHUBTRANSPORT_34_014: [If the provided connectionId is associated with the current connection, This function shall invoke updateStatus with status CONNECTED, change
-            // reason CONNECTION_OK and a null throwable.]
             this.updateStatus(IotHubConnectionStatus.CONNECTED, IotHubConnectionStatusChangeReason.CONNECTION_OK, null);
         }
     }
 
+    @Override
+    public void onMultiplexedDeviceSessionEstablished(String connectionId, String deviceId)
+    {
+        if (connectionId.equals(this.iotHubTransportConnection.getConnectionId()))
+        {
+            log.debug("The device session in the multiplexed connection to the IoT Hub has been established for device {}", deviceId);
+            this.updateStatus(IotHubConnectionStatus.CONNECTED, IotHubConnectionStatusChangeReason.CONNECTION_OK, null, deviceId);
+        }
+    }
+
+    @Override
+    public void onMultiplexedDeviceSessionLost(Throwable e, String connectionId, String deviceId)
+    {
+        if (connectionId.equals(this.iotHubTransportConnection.getConnectionId()))
+        {
+            log.debug("The device session in the multiplexed connection to the IoT Hub has been lost for device {}", deviceId);
+            if (e == null)
```

Comment:
it is safe to assume that network or service related disconnection would always be accompanied by by a throwable, correct?

Reply:
Sort of. Sometimes the connection is lost without any meaningful context other than the event itself, but we do build a default throwable in cases like that

Reply:
For the graceful close case, there will be no throwable though